File: Lab4/main/java/window/controller/Controller.java
package window.controller;

import window.model.*;
import window.view.WindowView;

import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Controller {
    private WindowService model;
    private WindowView view;

    public Controller(WindowService model, WindowView view) {
        this.model = model;
        this.view = view;
    }

    public void userInput() {
        boolean condition = true;
        while (condition) {
            view.printStartMenu();
            Scanner in = new Scanner(System.in);
            int num = in.nextInt();
            Receiver receiver = new Receiver(model);
            if (num == 1) selfCreation(receiver);
            else if (num == 2) loremCreation(receiver);
            else if (num == 3) openWindow(receiver);
            else if (num == 4) proxyExample(receiver);
            else if (num == 0) {
                closeWindow(receiver);
                condition = false;
            } else view.printErrorMessage();
        }
    }

    public void proxyExample(Receiver rec) {
        Invoker inv = new Invoker(CommandFactory.getCommand("proxy", Arrays.asList(rec, Color.Blue)));
        inv.execute();
    }

    public void selfCreation(Receiver rec) {
        view.printSelfCreation();
        Scanner in = new Scanner(System.in);
        String str = in.nextLine();
        Invoker inv = new Invoker(CommandFactory.getCommand("counter", Arrays.asList(rec, str)));
        inv.execute();
    }

    public void loremCreation(Receiver rec) {
        view.printLoremCreation();
        Scanner in = new Scanner(System.in);
        int num = in.nextInt();
        WindowGenerator gen = WindowGenerator.getInstance();
        String str = gen.generate(num);
        Invoker inv = new Invoker(CommandFactory.getCommand("counter", Arrays.asList(rec, str)));
        inv.execute();
    }

    public void openWindow(Receiver rec) {
        Invoker inv = new Invoker(CommandFactory.getCommand("open", List.of(rec)));
        inv.execute();
    }

    public void closeWindow(Receiver rec) {
        Invoker inv = new Invoker(CommandFactory.getCommand("close", List.of(rec)));
        inv.execute();
    }

}

File: Lab6/src/main/kotlin/controller/Controller.kt
package controller

import model.WindowGenerator
import view.WindowView
import javax.inject.Inject

class Controller {
    @Inject
    lateinit var receiver: Receiver
    @Inject
    lateinit var view: WindowView
    init {
        DaggerApplicationComponent.create().inject(this)
    }
    fun userInput() {
        var condition = true
        while (condition) {
            view.printStartMenu()
            when (readLine()) {
                "1" -> selfCreation()
                "2" -> loremCreation()
                "3" -> openWindow()
                "0" -> {
                    closeWindow()
                    condition = false
                }
                else -> view.printErrorMessage()
            }
        }
    }

    private fun closeWindow() {
        val invoker = Invoker(CommandFactory.getCommand("close", listOf(receiver)))
        invoker.execute()
    }

    private fun openWindow() {
        val invoker = Invoker(CommandFactory.getCommand("open", listOf(receiver)))
        invoker.execute()
    }

    private fun loremCreation() {
        view.printLoremCreation()
        val num = readLine()?.toInt()
        val str = num?.let { WindowGenerator.generate(it) }
        val invoker = Invoker(CommandFactory.getCommand("counter", listOf(receiver, str)))
        invoker.execute()
    }

    private fun selfCreation() {
        view.printSelfCreation()
        val str = readLine()
        val invoker = Invoker(CommandFactory.getCommand("counter", listOf(receiver, str)))
        invoker.execute()
    }
}